# Use cpu.h from libwebp 1.5.0 to fix definition of AVX2 related stuff.
# 
# Without this fix, all intel archs (x86, x64, evansport) fail to build
# due lacking definition of _mm256_cvtsi256_si32 and _mm256_storeu2_m128i
# 
--- src/dsp/cpu.h.orig	2025-07-08 02:20:00.000000000 +0200
+++ src/dsp/cpu.h	2024-12-20 02:17:50.000000000 +0100
@@ -56,11 +56,6 @@
     (defined(_M_X64) || defined(_M_IX86))
 #define WEBP_MSC_SSE41  // Visual C++ SSE4.1 targets
 #endif
-
-#if defined(_MSC_VER) && _MSC_VER >= 1700 && \
-    (defined(_M_X64) || defined(_M_IX86))
-#define WEBP_MSC_AVX2  // Visual C++ AVX2 targets
-#endif
 #endif
 
 // WEBP_HAVE_* are used to indicate the presence of the instruction set in dsp
@@ -85,16 +80,6 @@
 #define WEBP_HAVE_SSE41
 #endif
 
-#if (defined(__AVX2__) || defined(WEBP_MSC_AVX2)) && \
-    (!defined(HAVE_CONFIG_H) || defined(WEBP_HAVE_AVX2))
-#define WEBP_USE_AVX2
-#endif
-
-#if defined(WEBP_USE_AVX2) && !defined(WEBP_HAVE_AVX2)
-#define WEBP_HAVE_AVX2
-#endif
-
-#undef WEBP_MSC_AVX2
 #undef WEBP_MSC_SSE41
 #undef WEBP_MSC_SSE2
 
